### PR TITLE
Yiz/grpc domain logout

### PR DESCRIPTION
**Pull request by sleepyzzzzzz**
## Changes made in this PR
add LogOut implementation + specs in grpc & domain layers

## Corresponding noted Ruby branch
https://github.com/ParamountDataManagement/paramount-api/pull/1960

## TODO - changes that needs to be made
< leave the todos here to help the reviewers know what's the remaining pieces that haven't been implemented >
                                                                                        
## Implementation Checklist
Please check following check list to ensure the tasks are considered and completed in this branch.

#### API Design
- [x] Defined gRPC API in proto file
- [x] regenerated go code using `make proto_generate` command

#### API implementation
- [x] Implemented API in corresponding gRPC server
- [x] Added tests in gRPC layer

#### Domain Service and DB query Implementation
- [ ] Added new query in dbstore layer
- [ ] Added tests in db store layer
                                                                                       
- [ ] Implemented functionality in Domain Service Layer
- [ ] Added test in Domain Service Layer

- [ ] Created a new branch in paramount-api, with comments there indicating the functionality in Ruby that's re-implemented in Golang

#### Postgres Layer - identify and potentially add db migration to create index for fast querying.
- [ ] Verified and potentially created index for the newly added query.

### CODE REVIEW

File: internal/domain/users/domainService.go [hunk 1]
```diff
@@ -262,6 +262,11 @@ func (s *DomainService) confirm(ctx context.Context, userInfo *UserInfo, confirm
 	return nil
 }
 
+// Logout invalidates a user access token, which sets it to be expired
```
Comment (lhan-the-real) [inline internal/domain/users/domainService.go:265]: Love this comment!

File: apis/grpcservice/grpcusers/usersGRPCServer_test.go [hunk 2]
```diff
@@ -102,4 +102,30 @@ var _ = Describe("UsersGRPCServer", func() {
 		Entry("should return false with valid email address format but no record in db", "fakeemail@fake.ai", &usersv1.HasUserResponse{HasUser: false}, false),
 	)
 
+	Describe("LogOut", func() {
+		Context("request is missing fields", func() {
+			It("shoudl return error with empty user token", func() {
```
Comment (lhan-the-real) [inline apis/grpcservice/grpcusers/usersGRPCServer_test.go:107]: nit `It("should return error`

**Review by lhan-the-real — APPROVED** [review]

File: apis/grpcservice/grpcusers/usersGRPCServer_test.go (hunk 2 above)
Comment (sleepyzzzzzz) [reply apis/grpcservice/grpcusers/usersGRPCServer_test.go:107]: fix

**Comment by sleepyzzzzzz** [comment]
add logout implementation + specs in grpc and domain layers

**Pull request merged**